### PR TITLE
Feature get endpoint

### PR DESCRIPTION
**Pull request by Sarahtonein**
Basic get endpoint has been created and tested.

Return data example:

[
	{
		"ID": 1,
		"CreatedAt": "2023-08-28T01:59:10.052530592Z",
		"UpdatedAt": "2023-08-29T03:07:53.339416578Z",
		"DeletedAt": null,
		"UserID": "ccf965ee-cb7a-4044-b395-87c0b7dcf3a7",
		"name": "boblane",
		"layer": "",
		"ens": "",
		"active": true,
		"states": {
			"ids": ""
		},
		"transform": {
			"RotationID": 0,
			"ScaleID": 0,
			"PositionID": 0,
			"position": {
				"ID": 0,
				"CreatedAt": "0001-01-01T00:00:00Z",
				"UpdatedAt": "0001-01-01T00:00:00Z",
				"DeletedAt": null,
				"x": 0,
				"y": 0,
				"z": 0
			},
			"rotation": {
				"ID": 0,
				"CreatedAt": "0001-01-01T00:00:00Z",
				"UpdatedAt": "0001-01-01T00:00:00Z",
				"DeletedAt": null,
				"x": 0,
				"y": 0,
				"z": 0,
				"w": 0
			},
			"scale": {
				"ID": 0,
				"CreatedAt": "0001-01-01T00:00:00Z",
				"UpdatedAt": "0001-01-01T00:00:00Z",
				"DeletedAt": null,
				"x": 0,
				"y": 0,
				"z": 0
			}
		}
	},
	{
		"ID": 2,
		"CreatedAt": "2023-08-28T02:01:24.63738758Z",
		"UpdatedAt": "2023-08-28T02:01:24.63738758Z",
		"DeletedAt": null,
		"UserID": "630108f0-eed5-4041-b57a-5a9fc5149dcc",
		"name": "gumbo421",
		"layer": "",
		"ens": "",
		"active": true,
		"states": {
			"ids": ""
		},
		"transform": {
			"RotationID": 1,
			"ScaleID": 1,
			"PositionID": 1,
			"position": {
				"ID": 1,
				"CreatedAt": "2023-08-28T02:01:24.625849766Z",
				"UpdatedAt": "2023-08-28T05:43:35.120674543Z",
				"DeletedAt": null,
				"x": 4,
				"y": 6,
				"z": 155
			},
			"rotation": {
				"ID": 1,
				"CreatedAt": "2023-08-28T02:01:24.631982802Z",
				"UpdatedAt": "2023-08-28T02:01:24.631982802Z",
				"DeletedAt": null,
				"x": 4,
				"y": 5,
				"z": 6,
				"w": 7
			},
			"scale": {
				"ID": 1,
				"CreatedAt": "2023-08-28T02:01:24.635304151Z",
				"UpdatedAt": "2023-08-28T02:01:24.635304151Z",
				"DeletedAt": null,
				"x": 1,
				"y": 1,
				"z": 1
			}
		}
	}

### CODE REVIEW

**Comment by Balugrizzly** [comment]
> Basic get endpoint has been created and tested.
> 
> Return data example:
> 
> [ { "ID": 1, "CreatedAt": "2023-08-28T01:59:10.052530592Z", "UpdatedAt": "2023-08-29T03:07:53.339416578Z", "DeletedAt": null, "UserID": "ccf965ee-cb7a-4044-b395-87c0b7dcf3a7", "name": "boblane", "layer": "", "ens": "", "active": true, "states": { "ids": "" }, "transform": { "RotationID": 0, "ScaleID": 0, "PositionID": 0, "position": { "ID": 0, "CreatedAt": "0001-01-01T00:00:00Z", "UpdatedAt": "0001-01-01T00:00:00Z", "DeletedAt": null, "x": 0, "y": 0, "z": 0 }, "rotation": { "ID": 0, "CreatedAt": "0001-01-01T00:00:00Z", "UpdatedAt": "0001-01-01T00:00:00Z", "DeletedAt": null, "x": 0, "y": 0, "z": 0, "w": 0 }, "scale": { "ID": 0, "CreatedAt": "0001-01-01T00:00:00Z", "UpdatedAt": "0001-01-01T00:00:00Z", "DeletedAt": null, "x": 0, "y": 0, "z": 0 } } }, { "ID": 2, "CreatedAt": "2023-08-28T02:01:24.63738758Z", "UpdatedAt": "2023-08-28T02:01:24.63738758Z", "DeletedAt": null, "UserID": "630108f0-eed5-4041-b57a-5a9fc5149dcc", "name": "gumbo421", "layer": "", "ens": "", "active": true, "states": { "ids": "" }, "transform": { "RotationID": 1, "ScaleID": 1, "PositionID": 1, "position": { "ID": 1, "CreatedAt": "2023-08-28T02:01:24.625849766Z", "UpdatedAt": "2023-08-28T05:43:35.120674543Z", "DeletedAt": null, "x": 4, "y": 6, "z": 155 }, "rotation": { "ID": 1, "CreatedAt": "2023-08-28T02:01:24.631982802Z", "UpdatedAt": "2023-08-28T02:01:24.631982802Z", "DeletedAt": null, "x": 4, "y": 5, "z": 6, "w": 7 }, "scale": { "ID": 1, "CreatedAt": "2023-08-28T02:01:24.635304151Z", "UpdatedAt": "2023-08-28T02:01:24.635304151Z", "DeletedAt": null, "x": 1, "y": 1, "z": 1 } } }

Nice Fmting mate :)

**Comment by Balugrizzly** [comment]
@laprasdev Do you want to get inactive players here?
it currently only sends active ones back.

**Comment by Balugrizzly** [comment]
Will merge this After the monday multiplayer meetings

**Comment by laprasdev** [comment]
@Balugrizzly @Sarahtonein could we pass something like `include_active=true` in query params?

**Comment by laprasdev** [comment]
@Sarahtonein may i recommend triple ticks :) \```

```
like this
```

**Comment by Sarahtonein** [comment]
@laprasdev , should this pr be closed and a new one opened for branch feature_crud_requests?

**Comment by laprasdev** [comment]
I would defer to balu in the future but Id probably leave it as is for now

File: controllers/rest_get_player.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	db := database.GetDB()
+
+	// Read the "active" query parameter from the URL
+	activeParam := c.QueryParam("include_active")
+
+	// Initialize a variable to store the filter value
+	var active bool
+
+	// Check if the "active" query parameter is provided and parse it as a boolean
+	switch activeParam {
```
Comment (Balugrizzly) [inline controllers/rest_get_player.go:25]: @Sarahtonein 

May i suggest next time using:

<b, err := strconv.ParseBool(str)>

**Comment by Balugrizzly** [comment]
> @laprasdev , should this pr be closed and a new one opened for branch feature_crud_requests?

PR should remain open and be merged 
new branch should be created for other features

**Comment by Balugrizzly** [comment]
@Sarahtonein Will keep this branch not deleted for ease of reading the replies for you 
Once you did you can delete it

**Pull request merged**